CONFIGURATION.md

```diff
@@ -114,6 +114,7 @@ rebalance_cb                             |  C  |                 |
 offset_commit_cb                         |  C  |                 |               | low        | Offset commit result propagation callback. (set with rd_kafka_conf_set_offset_commit_cb()) <br>*Type: see dedicated API*
 enable.partition.eof                     |  C  | true, false     |         false | low        | Emit RD_KAFKA_RESP_ERR__PARTITION_EOF event whenever the consumer reaches the end of a partition. <br>*Type: boolean*
 check.crcs                               |  C  | true, false     |         false | medium     | Verify CRC32 of consumed messages, ensuring no on-the-wire or on-disk corruption to the messages occurred. This check comes at slightly increased CPU usage. <br>*Type: boolean*
+allow.auto.create.topics                 |  C  | true, false     |         false | low        | Allow automatic topic creation on the broker when subscribing to or assigning non-existent topics. The broker must also be configured with `auto.create.topics.enable=true` for this configuraiton to take effect. Note: The default value (false) is different from the Java consumer (true). Requires broker version >= 0.11.0.0, for older broker versions only the broker configuration applies. <br>*Type: boolean*
```

Comment:
technically requires a major version bump... but probably not important.

Reply:
Theoretically yes, but in practice there is no difference between having a consumer auto-create a topic which is later produced to by a producer, or the consumer subscribing to a non-existent topic until the producer creates it (automatically) when producing to it.
In short, auto topic creation doesn't make much sense in the consumer since the topic will be empty anyway.

This is also the reason we have a different default than the Java consumer. I asked Ismael about it and he agreed that our default here makes more sense.